### PR TITLE
MBS-12155: Filter out downvoted tags from JSON WS user-tags/genres

### PR DESCRIPTION
**Pull request by reosarevok**
### Fix MBS-12155

These are already filtered out in `_serialize_user_tag` and `_serialize_user_genre` in `XMLSerializer`, but they were not being looked into in the JSON serializer. This just adds an extra `grep` for `is_upvote` to `serialize_tags` to effectively do the same there.

For some reason, we had no tests at all for JSON user-tags, too. Added a JSON version of some of the ones we have for XML, and modified the test data so that it will fail if this bug is broken again.

### CODE REVIEW

File: lib/MusicBrainz/Server/WebService/Serializer/JSON/2/Utils.pm
Comment (yvanzo) [inline lib/MusicBrainz/Server/WebService/Serializer/JSON/2/Utils.pm:328]: Coding style looks weird. Personally I would always indent. Not sure about line returns and `,` separator.
@mwiencek: any suggestion?

**Review by yvanzo — APPROVED** [review]
LGTMBDNT. Thanks for the tests. Would like to have a second review on this.

File: lib/MusicBrainz/Server/WebService/Serializer/JSON/2/Utils.pm
Comment (mwiencek) [reply lib/MusicBrainz/Server/WebService/Serializer/JSON/2/Utils.pm:328]: I guess I would indent the `grep` since `@{ $opts->{user_genres} }` is indented, and it's a continuation of the `map`.  The comma is needed because there are two forms of `map`: https://perldoc.perl.org/functions/map and here it's using `map EXPR,LIST`.  Since it's returning a hash ref, it avoids double brackets in the other form like `map { { id => ... } } @list`.

**Review by mwiencek — APPROVED** [review]

**Pull request merged**
